### ISSUE TITLE
Run a private PoA network that supports Shanghai/PUSH0 without --dev

### ISSUE THREAD

**Issue by rmlearney-digicatapult** [ethereum/go-ethereum#28449]
Have been trying to fire up a private network based largely upon the Goerli `dumpgenesis` with Clique PoA

If I keep the following lines in the genesis that make the network think it's post-merge (so I can use PUSH0), it switches to PoS and searches for a beacon chain before falling over quite quickly:

```json
        "shanghaiTime": 0,
        "terminalTotalDifficulty": 0,
        "terminalTotalDifficultyPassed": true,
```

If I remove these options, then the network remains PoA but PUSH0 does not work (i.e. London)

I note a similar issue was raised under #27327 which resulted in code changes, which fires up a simulated beacon chain by using the `--dev` switch. But this is PoS and not PoA.

Is it possible to run a private PoA network that is compatible with Shanghai/PUSH0?

**Comment by q9f** [comment]
Currently, you only have two options:

- either use the `dev` network which mocks consensus and gives you a workable dev environment
- or setup your own merged network with a real consensus layer
  - to understand it, read [How to merge an Ethereum network right from the genesis block](https://dev.to/q9/how-to-merge-an-ethereum-network-right-from-the-genesis-block-3454)
  - or go strait to take a look at [pari's ansible playbooks](https://github.com/ethereumjs/consensus-deployment-ansible)

To make the beacon chain permissioned (as in PoA), you would have to modify the deposit contract (as [Sepolia](https://github.com/eth-clients/sepolia) does). But I'm not sure the Geth team can help you with that, it's pretty much out of scope for the client development.

Hope that helps.

**Comment by karalabe** [comment]
> Is it possible to run a private PoA network that is compatible with Shanghai/PUSH0?

Unfortunately not. Clique was a very simple change of PoW/Ethash to make it work with signatures. The switch to PoS introduced a lot of new changes that cannot be backported into Clique itself.

Whilst the PUSH0 might be doable, that would mean forks mean different things on different networks, which we are against with Geth. We've always committed to follow the Ethereum specs, so any network based off of Geth needs to do the same to.

TL;DR: You need a merged private network with all the bells and whistels.
